apps/src/music/views/SoundsPanel2.tsx

```diff
@@ -0,0 +1,487 @@
+    currentFolderRef.current = ref;
+  };
+
+  const currentSoundRefCallback = (ref: HTMLDivElement) => {
+    currentSoundRef.current = ref;
+  };
+
+  let possibleSoundEntries: SoundEntry[] = [];
+  let rightColumnSoundEntries: SoundEntry[] = [];
+
+  if (mode === 'packs') {
+    folders.sort((a, b) =>
+      a.restricted === b.restricted ? 0 : a.restricted ? -1 : 1
+    );
+    possibleSoundEntries = selectedFolder.sounds.map(sound => ({
+      folder: selectedFolder,
+      sound,
+    }));
+  } else {
+    folders.forEach(folder => {
+      folder.sounds.forEach(sound => {
+        possibleSoundEntries.push({folder, sound});
+      });
+    });
+  }
+
+  if (filter === 'all') {
+    rightColumnSoundEntries = possibleSoundEntries.filter(
+      soundEntry => soundEntry.sound.type !== 'preview'
+    );
+  } else {
+    rightColumnSoundEntries = possibleSoundEntries.filter(
+      soundEntry =>
+        soundEntry.sound.type === filter && soundEntry.sound.type !== 'preview'
+    );
+  }
+
+  const availableSoundTypes: {[key: string]: boolean} = {
+    all: true,
+    ...library.getAvailableSoundTypes(),
+  };
+
+  const allFilterButtons = [
+    {label: musicI18n.soundsFilterAll(), value: 'all'},
+    {label: musicI18n.soundsFilterBeats(), value: 'beat'},
+    {label: musicI18n.soundsFilterBass(), value: 'bass'},
+    {label: musicI18n.soundsFilterLeads(), value: 'lead'},
+    {label: musicI18n.soundsFilterEffects(), value: 'fx'},
+    {label: musicI18n.soundsFilterVocals(), value: 'vocal'},
+  ];
+
+  const filterButtons = allFilterButtons.filter(
+    filterButton => availableSoundTypes[filterButton.value]
+  );
+
+  return (
+    <FocusLock>
+      <div id="sounds-panel" className={styles.soundsPanel} aria-modal>
+        <div id="hidden-item" tabIndex={0} role="button" />
+        {showSoundFilters && (
+          <div id="sounds-panel-top" className={styles.soundsPanelTop}>
+            <SegmentedButtons
+              selectedButtonValue={mode}
+              buttons={[
+                {label: musicI18n.soundsFilterPacks(), value: 'packs'},
+                {label: musicI18n.soundsFilterSounds(), value: 'sounds'},
+              ]}
+              onChange={value => onModeChange(value as Mode)}
+              className={styles.segmentedButtons}
+            />
+
+            <SegmentedButtons
+              selectedButtonValue={filter}
+              buttons={filterButtons}
+              onChange={value => onFilterChange(value as Filter)}
+              className={styles.segmentedButtons}
+            />
+          </div>
+        )}
+        <div id="sounds-panel-body" className={styles.soundsPanelBody}>
+          {mode === 'packs' && (
+            <ScrollIntoView
+              id="sounds-panel-left"
+              className={styles.leftColumn}
+              doScroll={isDefaultSoundSelected}
+              delay={0}
+              frames={50}
+              distanceY={70}
```

Comment:
nit: constants for these (since we presumably want the same animation for both <ScrollIntoView> containers)?

Reply:
Good call; done.  The animation is almost the same, though with a slight delay for the sounds scroll, to illustrate that the two sections can scroll independently.